*Aula 05 Inferência Causal*

Vamos utilizar os dados do artigo de Cesar Zucco (2013) para rodar um modelo tosco, como exemplo do que não fazer em uma pesquisa.

A base de dados se encontra no link x, e o dicionário dos dados pode ser encontrado [aqui](https://www.dropbox.com/s/anth1tqxwejqdr3/Replication%20Instructions.pdf?dl=0).

Queremos mostrar que o crescimento da população evangélica diminui a busca pelo bolsa família.

Primeiro tentaremos mostrar a relação entre os gastos do bolsa família e o crescimento da população envagélica.

Em seguida, acrescentaremos outras variáveis como controle do modelo para obter o resultado que estamos buscando.

In [1]:
import os
import pandas as pd
import numpy as np
from statsmodels.formula.api import ols

In [4]:
pbf = pd.read_csv("pbf_completo.csv", sep = ",")
pbf.head()

codeibge            mun state region  alcance_2002  alcance_2006  \
0   1100106  GUAJARA MIRIM    RO      N      0.069851      0.284035   
1   1100205    PORTO VELHO    RO      N      0.050576      0.194233   
2   1100122      JI PARANA    RO      N      0.084964      0.146875   
3   1100023      ARIQUEMES    RO      N      0.126863      0.191644   
4   1100049         CACOAL    RO      N      0.106128      0.286071   

   alcance_2010  dgdpH_1_2002  dgdpH_1_2006  dgdpH_1_2010  ...  valid_2002  \
0      0.338100      0.123550     -0.089817     -0.169401  ...     15786.0   
1      0.201159      0.076698     -0.055391      0.205063  ...    164378.0   
2      0.216012      0.041789     -0.076465     -0.050747  ...     51866.0   
3      0.209192      0.004720     -0.033059     -0.054821  ...     34556.0   
4      0.225393      0.083267     -0.089112     -0.095071  ...     33760.0   

   valid_2002_2  valid_2006  valid_2006_2  valid_2010  valid_2010_2  \
0       15131.0       18852         18400       19524         18097   
1      157383.0      191481        189616      216681        201885   
2       49556.0       55069         53954       58357         54032   
3       33591.0       40349         39437       43571         40877   
4       32414.0       38451         38390       41084         38176   

   valid_depfed_1998  valid_depfed_2002  valid_depfed_2006  valid_depfed_2010  
0            13725.0            16863.0            19368.0            19616.0  
1           127284.0           165351.0           185556.0           210328.0  
2            41993.0            53440.0            54748.0            58699.0  
3            21185.0            36075.0            40697.0            43870.0  
4            31000.0            35068.0            37925.0            41120.0  

[5 rows x 70 columns]

In [5]:
list(pbf)

['codeibge',
 'mun',
 'state',
 'region',
 'alcance_2002',
 'alcance_2006',
 'alcance_2010',
 'dgdpH_1_2002',
 'dgdpH_1_2006',
 'dgdpH_1_2010',
 'dgdpH_2_2002',
 'dgdpH_2_2006',
 'dgdpH_2_2010',
 'dilma_vs_2010',
 'dilma_vs_2010_2',
 'distcap',
 'fam_2002_cct',
 'fam_2006_cct',
 'fam_2010_cct',
 'families_2002_ba',
 'families_2002_be',
 'fhc_vs_1998',
 'gini_2000',
 'hdi_2000',
 'incgov_2002',
 'incgov_2006',
 'incgov_2010',
 'lula_vs_1998',
 'lula_vs_2002',
 'lula_vs_2002_2',
 'lula_vs_2006',
 'lula_vs_2006_2',
 'mayorpty_2000',
 'mayorpty_2004',
 'mayorpty_2008',
 'nonwhite_2000',
 'nonwhite_2010',
 'pent_2000',
 'pent_2010',
 'pib_2002',
 'pib_2006',
 'pib_2009',
 'poor_covered_2002',
 'pop_2002',
 'pop_2006',
 'pop_2010',
 'psdb_vs_1998',
 'psdb_vs_2002',
 'pt_vs_1998',
 'pt_vs_2002',
 'pt_vs_2006',
 'pt_vs_2010',
 'serra_vs_2002',
 'serra_vs_2002_2',
 'spentH_2002_cct',
 'spentH_2006_cct',
 'spentH_2010_cct',
 'targetalcance_2006',
 'targetbf_2006',
 'tot_fam2002',
 'valid_2002',


In [7]:
# Montando a variável de crescimento pentecostal utilizando os dados do censo de 2000 e 2010.

pbf["pent_delta"] = pbf["pent_2010"] - pbf["pent_2000"]

In [8]:
# Criando o modelo de regressão simples, utilizando os gastos com bf dependente
# e crescimento da população evangélica (proporcional à população) como variável independente.

model_envang_1 = ols("spentH_2010_cct ~ pent_delta",
                     data = pbf)
m_declinio = model_envang_1.fit()
m_declinio.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        spentH_2010_cct   R-squared:                       0.001
Model:                            OLS   Adj. R-squared:                  0.001
Method:                 Least Squares   F-statistic:                     5.439
Date:                Sun, 14 Nov 2021   Prob (F-statistic):             0.0197
Time:                        19:06:20   Log-Likelihood:                -5389.1
No. Observations:                5505   AIC:                         1.078e+04
Df Residuals:                    5503   BIC:                         1.080e+04
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
Intercept      1.0725      0.012     91.137      0.000       1.049       1.096
pent_delta     0.0067      0.003      2.332      0.020       0.001       0.012
==============================================================================
Omnibus:                     7212.968   Durbin-Watson:                   0.531
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              389.794
Skew:                           0.199   Prob(JB):                     2.28e-85
Kurtosis:                       1.759   Cond. No.                         5.73
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

No modelo encontramos uma relação positiva, ou seja, o aumento da população evangálica e bolsa família possui uma correlação, o contrário do que quero mostrar.

p_valor = 0.020

Por uma convenção científica, p-valores maiores de 5% não são significantes e acima de 10% nem merecem menção. Ou seja, se tudo o que a gente precisasse era mostrar que não há uma relação entre PBF e pent_delta, nosso trabalho sujo já teria terminado!
 
Agora, adicionaremos uma variável de controle ao modelo.

Votação do Lula em 2002.

Podemos argumentar no nosso artigo que o governo petista busca ajudar com mais bolsa-familia lugares que votaram no Lula em 2002, e ao mesmo tempo uma votação expressiva do PT afugenta pentecostais. A gente pode sempre contar uma história.

In [9]:
# Adicionando Lula_vs_2002, votação no primeiro turno de 2006

model_envang_2 = ols("spentH_2010_cct ~ pent_delta + lula_vs_2002",
                     data = pbf)
m_declinio = model_envang_2.fit()
m_declinio.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        spentH_2010_cct   R-squared:                       0.096
Model:                            OLS   Adj. R-squared:                  0.096
Method:                 Least Squares   F-statistic:                     293.0
Date:                Sun, 14 Nov 2021   Prob (F-statistic):          1.18e-121
Time:                        19:28:24   Log-Likelihood:                -5113.2
No. Observations:                5505   AIC:                         1.023e+04
Df Residuals:                    5502   BIC:                         1.025e+04
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept        1.7798      0.031     56.628      0.000       1.718       1.841
pent_delta       0.0044      0.003      1.614      0.107      -0.001       0.010
lula_vs_2002    -1.6514      0.069    -24.084      0.000      -1.786      -1.517
==============================================================================
Omnibus:                     1320.165   Durbin-Watson:                   0.634
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              264.800
Skew:                           0.189   Prob(JB):                     3.16e-58
Kurtosis:                       1.994   Cond. No.                         37.8
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Ainda é positiva a relação, mas encontramos um p_valor de 0.107.

Vamos utilizar a votação do Lula em 2006, trocando a variável, vamos entender o que acontece com nossa relação.

In [10]:
# Adicionando lula_vs_2002, votação do Lula no primeiro turno de 2002.

model_envang_3 = ols("spentH_2010_cct ~ pent_delta + lula_vs_2006",
                    data = pbf)
m_declinio = model_envang_3.fit()
m_declinio.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        spentH_2010_cct   R-squared:                       0.578
Model:                            OLS   Adj. R-squared:                  0.578
Method:                 Least Squares   F-statistic:                     3767.
Date:                Sun, 14 Nov 2021   Prob (F-statistic):               0.00
Time:                        19:35:16   Log-Likelihood:                -3017.6
No. Observations:                5505   AIC:                             6041.
Df Residuals:                    5502   BIC:                             6061.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
================================================================================
                   coef    std err          t      P>|t|      [0.025      0.975]
--------------------------------------------------------------------------------
Intercept       -0.3236      0.018    -18.155      0.000      -0.359      -0.289
pent_delta      -0.0036      0.002     -1.946      0.052      -0.007    2.73e-05
lula_vs_2006     2.7515      0.032     86.722      0.000       2.689       2.814
==============================================================================
Omnibus:                       79.385   Durbin-Watson:                   1.349
Prob(Omnibus):                  0.000   Jarque-Bera (JB):               83.351
Skew:                           0.283   Prob(JB):                     7.95e-19
Kurtosis:                       3.206   Cond. No.                         26.5
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

Agora encontramos uma relação negativa entre o crescimento da população evangélica e o recebimento de bolsa família. Ou seja, quanto mais cresce a população evangálica, menos pessoas necessitam do auxílio.

pent_delta = -0.0036

p_valor = 0.052

Aqui, estamos mais próximos de mostrar o que propusemos inicialmente, porém, não atingimos um p_valor menor que 5%.

Por isos, vamos utilizar a votação no segundo turno no Lular em 2006.

In [11]:
model_envang_4 = ols("spentH_2010_cct ~ pent_delta + lula_vs_2006_2",
                    data = pbf)
m_declinio = model_envang_4.fit()
m_declinio.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:        spentH_2010_cct   R-squared:                       0.530
Model:                            OLS   Adj. R-squared:                  0.530
Method:                 Least Squares   F-statistic:                     3103.
Date:                Sun, 14 Nov 2021   Prob (F-statistic):               0.00
Time:                        19:40:40   Log-Likelihood:                -3313.3
No. Observations:                5505   AIC:                             6633.
Df Residuals:                    5502   BIC:                             6652.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
==================================================================================
                     coef    std err          t      P>|t|      [0.025      0.975]
----------------------------------------------------------------------------------
Intercept         -0.6219      0.023    -27.047      0.000      -0.667      -0.577
pent_delta        -0.0039      0.002     -1.963      0.050      -0.008   -4.36e-06
lula_vs_2006_2     2.7772      0.035     78.702      0.000       2.708       2.846
==============================================================================
Omnibus:                       14.716   Durbin-Watson:                   1.272
Prob(Omnibus):                  0.001   Jarque-Bera (JB):               14.742
Skew:                           0.125   Prob(JB):                     0.000629
Kurtosis:                       3.037   Cond. No.                         29.2
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

pent_delta = -0.0039

p_valor = 0.050


Em uma lista com 70 variáveis, selecionamos aquela que consegue provar o nosso ponto, estamos fazendo **p_hacking**. Fazendo pequenas mudanças no modelo, eu construo a relação que eu quero, utilizando, não necessariamente, coisas que realmente importam para o estudo ou para o problema de pesquisa.